aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+   </form>
+```
+
+Notice that the form action posts to the vulnerable site, not to the malicious site. This is the “cross-site” part of CSRF.
+
+4. The user clicks the submit button. The browser automatically includes the authentication cookie for the requested domain (the vulnerable site in this case) with the request.
+
+5. The request runs on the server with the user’s authentication context, and can do anything that an authenticated user is allowed to do.
+
+Although this example requires the user to click the form button, the malicious page could just as easily run a script (without any user interaction) that sends an AJAX request. Moreover, using SSL does not prevent a CSRF attack, because the malicious site can send an `https://` request. Some attacks can target site endpoints that respond to `GET` requests, in which case even an image tag can be used to perform the action (this form of attack is common on forum sites that permit images but block JavaScript). If your application uses `GET` requests to significantly change the state of the application, you should switch to `POST` if possible (in addition to protecting against CSRF attacks).
+
+Typically, CSRF attacks are possible against web sites that use cookies for authentication, because browsers send all relevant cookies to the destination web site. However, CSRF attacks are not limited to exploiting cookies. For example, Basic and Digest authentication are also vulnerable. After a user logs in with Basic or Digest authentication, the browser automatically sends the credentials until the session ends. Users can guard against CSRF vulnerabilities by being sure to log out of sites when they are done using them, and clearing their browser's cookies periodicially. However, CSRF vulnerabilities are fundamentally problems with the web app, not the end user, and addressing them in ASP.NET Core is described below.
+
+## How does ASP.NET Core MVC address CSRF?
+
+The most common approach to defending against CSRF attacks is the synchronizer token pattern (STP). STP is a technique where the server sends a token associated with the current user's session to the client with each request that contains a form, and the client must then send back the token to the server for verification. If the server receives a token that doesn't match the authenticated user's session, the request should be rejected. The token is unique and unpredictable, and can also be used to ensure proper sequencing of a series of requests (ensuring page 1 precedes page 2 which precedes page 3). ASP.NET Core MVC will generate Antiforgery Tokens by default on all forms it generates. The following two examples of view logic will generate antiforgery tokens automatically:
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```html
+@using (Html.BeginForm("ChangePassword", "Manage"))
```

Comment:
Product team discussion indicated to show TH, HTML Helpers, and direct razor inclusion.